test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -0,0 +1,403 @@
+      if (
+        seq !== preSetupSequencers.Ethereum &&
+        seq !== preSetupSequencers.Arbitrum
+      ) {
+        await leaveSequencingIfAlreadySequencer(seq);
+        anySequencerGone = true;
+      }
+    }
+  }
+  if (anySequencerGone) {
+    await waitForNBlocks(10);
+  }
+});
+
+describe("Update cancellation -", () => {
+  let chain: any;
+  let testUser1: User;
+  let testUser2: User;
+  let testUser2Address: string;
+  beforeEach(async () => {
+    chain = "Ethereum";
+    const notYetSequencer = await createACollatorUser();
+    testUser2 = await createACollatorUser();
+    const minToBeSequencer = await SequencerStaking.minimalStakeAmount();
+    await signTx(
+      await getApi(),
+      await SequencerStaking.provideSequencerStaking(
+        minToBeSequencer.addn(1000),
+        "Ethereum",
+      ),
+      notYetSequencer.keyRingPair,
+    ).then((events) => {
+      expectExtrinsicSucceed(events);
+    });
+    const sequencers = await SequencerStaking.activeSequencers();
+    expect(sequencers.toHuman().Ethereum).toContain(
+      notYetSequencer.keyRingPair.address,
+    );
+    await signTx(
+      await getApi(),
+      await SequencerStaking.provideSequencerStaking(
+        minToBeSequencer.addn(1000),
+        "Ethereum",
+      ),
+      testUser2.keyRingPair,
+    ).then((events) => {
+      expectExtrinsicSucceed(events);
+    });
+    testUser1 = notYetSequencer;
+    testUser2Address = testUser2.ethAddress.toString();
+    testUser1.addAsset(MGA_ASSET_ID);
+    testUser2.addAsset(MGA_ASSET_ID);
+  });
+  it("GIVEN a sequencer, WHEN <correctly> canceling an update THEN a % of the slash is given to it", async () => {
+    const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
+      testUser1,
+      testUser2,
+      chain,
+    );
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await Rolldown.waitForReadRights(testUser2Address, 50, "Ethereum");
+    const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    //we approve the cancellation
+    const cancelResolutionEvents = await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        testUser2Address,
+        new L2Update(api)
+          .withCancelResolution(txIndex, reqIdCanceled, true)
+          .on("Ethereum")
+          .build(),
+      ),
+    );
+    await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+    await testUser2.refreshAmounts(AssetWallet.BEFORE);
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await testUser2.refreshAmounts(AssetWallet.AFTER);
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+    const slashRewardCanceler = testUser2
+      .getAsset(MGA_ASSET_ID)
+      ?.amountAfter.free!.sub(
+        testUser2.getAsset(MGA_ASSET_ID)?.amountBefore.free!,
+      );
+    const slashFineUpdater = testUser1
```

Comment:
plz rename to something meaningful on this context

Reply:
renamed